pkg/deploy/controller/lifecycle/manager.go

```diff
@@ -0,0 +1,164 @@
+package lifecycle
+
+import (
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+)
+
+// Status represents the status of a lifecycle action, and should be treated
+// as the source of truth by controllers.
+type Status string
+
+const (
+	// Pending means the action has not yet executed.
+	Pending Status = "Pending"
```

Comment:
Here's an example. A pre-hook using execnewpod fails, and the hook is configured to ignore failures. The stored status in `PreExecNewPodActionPodPhaseAnnotation` is `PodFailed`. The "logical" status reported for this by the plugin via the lifecycle manager is "Complete" since the status is terminal and the failure is ignored.

Is having the plugin-specific `PodFailed` status enough to provide to users/UI, or does the logical status also need captured? The logical status seems almost like an implementation detail.